server/src/packages/landuse-form/landuse-actions/landuse-actions.controller.ts

```diff
@@ -29,16 +29,35 @@ export class LanduseActionsController {
   @Patch('/:id')
   async update(@Body() body, @Param('id') id) {
     const allowedAttrs = pick(body, LANDUSE_ACTION_ATTRS);
+    const dateOfPreviousApproval = await this.setDate(body.dcp_dateofpreviousapproval);
+    const lapseDateOfPreviousApproval = await this.setDate(body.dcp_lapsedateofpreviousapproval);
+    const recordationDate = await this.setDate(body.dcp_recordationdate);
 
     await this.crmService.update('dcp_landuseactions', id, {
       ...allowedAttrs,
 
       ...(body.chosen_zoning_resolution_id ? { 'dcp_zoningresolutionsectionactionispursuantto@odata.bind': `/dcp_zoningresolutions(${body.chosen_zoning_resolution_id})` } : {}),
+      dcp_dateofpreviousapproval: dateOfPreviousApproval,
+      dcp_lapsedateofpreviousapproval: lapseDateOfPreviousApproval,
+      dcp_recordationdate: recordationDate,
     });
 
     return {
       dcp_landuseactionid: id,
       allowedAttrs,
     };
   }
+
+  // when a user selects a date from the frontend date-picker, this value
+  // is sent to backend as a datestring in an array, so we need to grab
+  // the first item in that array. When the user does not use the date-picker,
+  // and instead the date value comes from CRM (still exists in the body), the
+  // date is just formatted as an isolated datestring and does not need to be reformatted.
+  async setDate(bodyDateField) {
+    if (bodyDateField) {
+      return Array.isArray(bodyDateField) ? bodyDateField[0] : bodyDateField;
+    } else {
+      return null;
```

Comment:
comment [A]
hmm... it looks like if one of the date fields, say `dcp_dateofpreviousapproval`, isn't even in the body, then this function returns null. What if this overwrites a value in CRM? Are we assuming that if a value exists in CRM, then bodyDateField would never be null?

Reply:
these date fields are always included in the body regardless of whether the user has updated them manually or not. So if the user did not change anything, what's in the body would be what was saved in CRM previously.

Reply:
ahh ok. Got you. Okay if that holds true then sounds good for now.
Just a note: In general I think it's better to program a little defensively just in case this endpoint is used differently. 
For example, if someone sent a custom PATCH request not through Ember, but through Postman and forgot to include `dcp_dateofpreviousapproval`, then the existing value in CRM would be overwritten. Just something to keep in mind!